compiler/rustc_codegen_llvm/src/attributes.rs

```diff
@@ -317,7 +317,7 @@ pub fn from_fn_attrs(cx: &CodegenCx<'ll, 'tcx>, llfn: &'ll Value, instance: ty::
     // Note that currently the `wasm-import-module` doesn't do anything, but
     // eventually LLVM 7 should read this and ferry the appropriate import
     // module to the output file.
-    if cx.tcx.sess.target.arch == "wasm32" {
+    if cx.tcx.sess.target.arch.starts_with("wasm") {
```

Comment:
This should probably add a target property `is_like_wasm` or similar, much like e.g. `is_like_emscripten`, `is_like_windows` and similar.